src/Xamarin.Android.Build.Tasks/Tasks/JdkInfo.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Xamarin.Android.Tools;
+
+namespace Xamarin.Android.Tasks
+{
+	public class JdkInfo : Task
+	{
+		[Output]
+		public string JdkJvmPath { get; set; }
+
+		public override bool Execute ()
+		{
+			Log.LogMessage (MessageImportance.Low, $"Task {nameof (JdkInfo)}");
+
+			var jdk = Xamarin.Android.Tools.JdkInfo.GetKnownSystemJdkInfos (CreateTaskLogger (this)).FirstOrDefault ();
+
+                        if (jdk == null) {
+                                Log.LogError ("Could not determine JAVA_HOME location. Please set JdksRoot or export the JAVA_HOME environment variable.");
+                                return false;
+                        }
+
+			try {
+				var jdkJvmPath = jdk.JdkJvmPath;
+				if (string.IsNullOrEmpty(jdkJvmPath)) {
+					Log.LogError ("JdkJvmPath is blank");
+					return false;
+				}
+
+				Log.LogMessage (MessageImportance.Low, $"  {nameof (jdk.JdkJvmPath)}: {jdk.JdkJvmPath}");
+
+				if (!File.Exists (jdkJvmPath)) {
+					Log.LogError ($"JdkJvmPath not found at {jdkJvmPath}");
+					return false;
+				}
+
+				JdkJvmPath = jdkJvmPath;
+				Log.LogMessage (MessageImportance.Low, $"  [Output] {nameof (JdkJvmPath)}: {JdkJvmPath}");
+
+				return !Log.HasLoggedErrors;
+			}
+			finally {
+			}
+		}
+
+		static Action<TraceLevel, string> CreateTaskLogger (Task task)
```

Comment:
I already advocated for the removal of this task, but we already have this method, in `Utilities/MSBuildExtensions.cs`, as a `.CreateTaskLogger()` extension method.